src/symbols/Node.tsx

```diff
@@ -123,60 +132,39 @@ export const Node: FC<NodeProps> = ({
   useCursor(isDragging, 'grabbing');
 
   return (
-    <a.group ref={group} position={nodePosition as any} {...(bind() as any)}>
-      {icon ? (
-        <Icon
-          id={id}
-          image={icon}
-          size={nodeSize + 8}
-          opacity={selectionOpacity}
-          animated={animated}
-          onClick={() => {
-            if (!disabled && !isDragging) {
-              onClick?.(node);
-            }
-          }}
-          onPointerOver={() => onPointerOver?.(node)}
-          onPointerOut={() => onPointerOut?.(node)}
-          onActive={setActive}
-          onContextMenu={() => {
-            if (!disabled) {
-              setMenuVisible(true);
-              onContextMenu?.(node);
-            }
-          }}
-        />
-      ) : (
-        <Sphere
-          id={id}
-          size={nodeSize}
-          color={
-            isSelected || active || isDragging || isActive
-              ? theme.node.activeFill
-              : fill || theme.node.fill
-          }
-          opacity={selectionOpacity}
-          animated={animated}
-          onPointerOver={() => onPointerOver?.(node)}
-          onPointerOut={() => onPointerOut?.(node)}
-          onClick={() => {
-            if (!disabled && !isDragging) {
-              onClick?.(node);
-            }
-          }}
-          onActive={val => {
-            if (!disabled && !panning) {
-              setActive(val);
-            }
-          }}
-          onContextMenu={() => {
-            if (!disabled) {
-              setMenuVisible(true);
-              onContextMenu?.(node);
-            }
-          }}
-        />
-      )}
+    <a.group
+      ref={group}
+      position={nodePosition as any}
+      {...(bind() as any)}
+      onClick={() => {
+        if (!disabled) {
+          onClick?.(node);
+        }
+      }}
+      onContextMenu={() => {
+        if (!disabled) {
+          setMenuVisible(true);
+          onContextMenu?.(node);
+        }
+      }}
+      onPointerOver={() => {
+        onPointerOver?.(node);
+        if (!disabled && !panning) setActive(true);
+      }}
+      onPointerOut={() => {
+        onPointerOut?.(node);
+        setActive(false);
+      }}
+    >
+      {renderNode({
```

Comment:
So would this be a breaking change to remove the `icon` property from the `node` model?